File: Study/implementation/C/02.c
#include <stdio.h>
#include <string.h>

int main() {
  int n, i, sum, count = 0;
  char hour[3], minute[3], second[3];

  scanf("%d", &n);

  sum = n * 3600 + 3599;

  for (i = 0; i < sum + 1; i++) {
    sprintf(hour, "%d", i / 3600);
    sprintf(minute, "%d", i % 3600 / 60);
    sprintf(second, "%d", i % 60);
    if (
      strstr(hour, "3") != NULL ||
      strstr(minute, "3") != NULL ||
      strstr(second, "3") != NULL
    ) {
      count += 1;
    }
  }

  printf("%d", count);

  return 0;
}

File: Study/implementation/Python/02.py
n = int(input())
sum = n * 3600 + 3599
count = 0

for i in range(sum + 1):
  hour = str(i // 3600)
  minute = str(i % 3600 // 60)
  second = str(i % 60)
  if '3' in hour + minute + second:
    count += 1

print(count)
